tripper/namespace.py

```diff
@@ -0,0 +1,181 @@
+"""Provides a simple representation of namespaces."""
+from .errors import NamespaceError, NoSuchIRIError
+
+
+class Namespace:
+    """Represent a namespace.
+
+    Arguments:
+        iri: IRI of namespace to represent.
+        label_annotations: Sequence of label annotations. If given, check
+            the underlying ontology during attribute access if the name
+            correspond to a label. The label annotations should be ordered
+            from highest to lowest precedense.
+            If True is provided, `label_annotations` is set to
+            ``(SKOS.prefLabel, RDF.label, SKOS.altLabel)``.
+        check: Whether to check underlying ontology if the IRI exists during
+            attribute access.  If true, NoSuchIRIError will be raised if the
+            IRI does not exist in this namespace.
+        cachemode: Should be one of:
+              - Namespace.NO_CACHE: Turn off caching.
+              - Namespace.USE_CACHE: Cache attributes as they are looked up.
+              - Namespace.ONLY_CACHE: Cache all names at initialisation time.
+                Do not access the triplestore after that.
+            Default is `NO_CACHE` if neither `label_annotations` or `check`
+            is given, otherwise `USE_CACHE`.
+        triplestore: Use this triplestore for label lookup and checking.
+            If not given, and either `label_annotations` or `check` are
+            enabled, a new rdflib triplestore will be created.
+        triplestore_url: Alternative URL to use for loading the underlying
+            ontology if `triplestore` is not given.  Defaults to `iri`.
+    """
+
+    NO_CACHE = 0
+    USE_CACHE = 1
+    ONLY_CACHE = 2
+
+    __slots__ = (
+        "_iri",
+        "_label_annotations",
+        "_check",
+        "_cache",
+        "_triplestore",
+    )
+
+    def __init__(
+        self,
+        iri,
+        label_annotations=(),
+        check=False,
+        cachemode=-1,
+        triplestore=None,
+        triplestore_url=None,
+    ):
+        if label_annotations is True:
```

Comment:
```suggestion
        if label_annotations is not None:
```

Reply:
The value `True` is explicitly stated in the docstring.